spec/hyacinth/adapters/preservation_adapter/fedora3/system_spec.rb

```diff
@@ -0,0 +1,172 @@
+        expect(rubydora_object.state).to eql('I')
+      end
+    end
+    context "basic rels-ext properties" do
+      let(:dsids) { ['structMetadata'] }
+      let(:digital_object_title) { "Assigned Label" }
+      let(:model_property) do
+        {
+          predicate: "info:fedora/fedora-system:def/model#hasModel",
+          object: "info:fedora/ldpd:ContentAggregator",
+          pid: object_pid
+        }
+      end
+      let(:rdftype_property) do
+        {
+          predicate: "http://www.w3.org/1999/02/22-rdf-syntax-ns#type",
+          object: "http://purl.oclc.org/NET/CUL/Aggregator",
+          pid: object_pid
+        }
+      end
+      let(:restriction_property) do
+        {
+          predicate: "http://www.loc.gov/premis/rdf/v1#hasRestriction",
+          object: "onsite restriction",
+          pid: object_pid,
+          isLiteral: true
+        }
+      end
+
+      it "persists model properties" do
+        hyacinth_object.dynamic_field_data['restrictions'] = { 'restricted_onsite' => true }
+        adapter.persist_impl("fedora3://#{object_pid}", hyacinth_object)
+      end
+    end
+    context "DC properties" do
+      # do not need to fetch profiles for DC and RELS-EXT
+
+      it "persists model properties" do
+        hyacinth_object.identifiers << 'keep'
+        adapter.persist_impl("fedora3://#{object_pid}", hyacinth_object)
+        expect(rubydora_object.datastreams['DC'].content).to include("<dc:identifier>keep</dc:identifier>")
+        expect(rubydora_object.datastreams['DC'].content).not_to include("<dc:identifier>remove</dc:identifier>")
+      end
+    end
+    context "Struct properties" do
+      # do not need to fetch profiles for DC and RELS-EXT
+      let(:dsids) { ['structMetadata'] }
+      let(:child_object_title) { "Assigned Label" }
+      let(:child_hyacinth_object) do
+        obj = DigitalObject::Item.new
+        obj.dynamic_field_data['title'] = [{ 'title_sort_portion' => child_object_title }]
+        obj.save
+        obj
+      end
+      let(:child_uid) { child_hyacinth_object.uid }
+
+      before do
+        hyacinth_object.structured_children['structure'] << child_uid
+      end
+
+      it "persists model properties" do
+        hyacinth_object.identifiers << 'keep'
+        adapter.persist_impl("fedora3://#{object_pid}", hyacinth_object)
+        # because .save is stubbed, the datastream should still be dirty
+        expect(rubydora_object.datastreams['structMetadata'].content).to include("<mets:div LABEL=\"#{child_object_title}\" ORDER=\"1\" CONTENTIDS=\"#{child_uid}\"/>")
+      end
+    end
+    context "RelsInt properties for resources" do
+      let(:dsids) { ['structMetadata'] }
+      let(:hyacinth_object) { DigitalObject::Asset.new }
+      let(:resource_args) { { original_filename: '/old/path/to/file.doc', location: '/path/to/file.doc', checksum: 'asdf', file_size: 'asdf' } }
+      before do
+        hyacinth_object.resources['master'].send(:initialize, resource_args)
```

Comment:
It looks like you're re-running the initialize method for an already-initialized object?  If this is a thing that people do, that's cool to know, but since I haven't seen it before it somehow feels wrong.  @cgalarza, out of curiosity, have you ever seen that done before?  If not, I thought you'd find this as interesting as I did.

Reply:
I mean, I don't recommend it in most cases - it's kind of a dirty trick